File: frontend-angular/src/app/interface/user-response-dto.ts
export interface UserResponseDTO {
  userId: string;
  email: string;
  firstName: string;
  lastName: string;
  role: string;
  phone: string;
  address?: string;
  profileImgUrl?: string;
  aboutMe?: string;
  country?: string;
  enabled: boolean;
  createdTime: string;
  accountStatus?: { [key: string]: boolean };
}

File: backend/src/main/java/com/elhabhab/backend/dto/response/UserResponseDTO.java
package com.elhabhab.backend.dto.response;


import com.elhabhab.backend.enums.UserRole;
import lombok.*;
import org.springframework.web.multipart.MultipartFile;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.Map;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserResponseDTO {
    private UUID userId;
    private String email;
    private String firstName;
    private String lastName;
    private String role;
    private String phone;
    private String address;
    private String profileImgUrl;
    private String aboutMe;
    private String country;
    private boolean enabled;
    private LocalDateTime createdTime;
    private Map<String, Boolean> accountStatus;
}
